pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.cpp

```diff
@@ -203,6 +203,157 @@ auto GateImplementationsLM::applyGeneratorSingleExcitationPlus(
     return -static_cast<PrecisionT>(0.5);
 }
 
+template <class PrecisionT>
+auto GateImplementationsLM::applyGeneratorDoubleExcitation(
+    std::complex<PrecisionT> *arr, std::size_t num_qubits,
+    const std::vector<std::size_t> &wires, [[maybe_unused]] bool adj)
+    -> PrecisionT {
+    using ComplexT = std::complex<PrecisionT>;
+    PL_ASSERT(wires.size() == 4);
+
+    const std::size_t rev_wire0 = num_qubits - wires[3] - 1;
+    const std::size_t rev_wire1 = num_qubits - wires[2] - 1;
+    const std::size_t rev_wire2 = num_qubits - wires[1] - 1;
+    const std::size_t rev_wire3 = num_qubits - wires[0] - 1;
+
+    const std::size_t rev_wire0_shift = static_cast<std::size_t>(1U)
+                                        << rev_wire0;
+    const std::size_t rev_wire1_shift = static_cast<std::size_t>(1U)
+                                        << rev_wire1;
+    const std::size_t rev_wire2_shift = static_cast<std::size_t>(1U)
+                                        << rev_wire2;
+    const std::size_t rev_wire3_shift = static_cast<std::size_t>(1U)
+                                        << rev_wire3;
+
+    auto parity = revWireParity(rev_wire0, rev_wire1, rev_wire2, rev_wire3);
+
+    for (std::size_t k = 0; k < exp2(num_qubits - 4); k++) {
+
```

Comment:
We should consider using an array of indices instead of stand-alone variables. We might be able to improve the codebase with such a change. Nothing for now, let's just keep it in mind.